### PR TITLE
Add snapshot tests to debian provider

### PR DESCRIPTION
**Pull request by willmurphyscode**
See #309 .

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -62,6 +66,18 @@ def metadata_schema_valid(self) -> bool:
 
         return True
 
+    def copy_input_fixtures(self, mock_data_path: str):
+        shutil.copytree(mock_data_path, self.input_dir, dirs_exist_ok=True)
+
+    def assert_result_snapshots(self):
+        for result_file in self.result_files():
```
Comment (willmurphyscode) [inline tests/conftest.py:73]: How does this guard against asserting just omitting some files from the result set?

File: tests/conftest.py [hunk 2]
```diff
@@ -102,20 +122,26 @@ def local_dir(self, path: str):
...
+
+        if input_fixture:
+            h.copy_input_fixtures(self.local_dir(input_fixture))
+
+        return h
 
 
 @pytest.fixture()
-def helpers(request, tmpdir):
+def helpers(request, tmpdir, snapshot):
```
[the remark this answers is not in the document]
Comment (wagoodman) [reply tests/conftest.py:140]: yeah, this is pytest fixture magic. 

Any test function or function that is decorated with `@pytest.fixture()` (is a test fixture) can get other test fixtures automatically dependency injected. The `pytest-snapshot` lib provides the `snapshot` test fixture which our fixture is leveraging in testing.

[the remark this answers is not in the document]
Comment (willmurphyscode) [reply tests/conftest.py:155]: That's a lot of magic, but I'm glad it keeps us from having to change every test that uses `Helpers`.

**Review by wagoodman — APPROVED** [review]
mind updating the DEVELOPING.md to hint how to update snapshots selectively for individual tests?

Otherwise 👍

**Pull request merged**